src/lamplib/src/genny/tasks/preprocess.py

```diff
@@ -322,12 +324,25 @@ def _replace_numexpr(self, input):
             raise ParseException(msg)
         except Exception as e:
             msg = (
-                "Failure trying to parse '{OP_KEY}' node.\n"
+                f"Failure trying to parse '{OP_KEY}' node.\n"
                 f"Node source: {input}\n"
                 f"Exception: {e}\n"
             )
             raise ParseException(msg)
 
+    def _replace_flattenarr(self, input):
+        OP_KEY = "^FlattenArray"
+
+        parsed_values = self._recursive_parse(input)
+
+        try:
+            return [value
+                    for parsed_value in parsed_values
+                    for value in (parsed_value if type(parsed_value) == list else [parsed_value])]
```

Comment:
I don't think this handles multiple levels of nesting, which I think you want.

Reply:
Ah, I can see how one might think that.

`^FlattenOnce` is closer to what I wanted to do. Automatically flattening all the subarrays seems like maybe an optional flag to toggle in the future.